src/app/modules/examples/__tests__/examples.routes.spec.ts

```diff
@@ -0,0 +1,655 @@
+import { errAsync } from 'neverthrow'
+import supertest, { Session } from 'supertest-session'
+import { mocked } from 'ts-jest/utils'
+
+import { IAgencySchema, IUserSchema } from 'src/types'
+
+import { createAuthedSession } from 'tests/integration/helpers/express-auth'
+import { setupApp } from 'tests/integration/helpers/express-setup'
+import { buildCelebrateError } from 'tests/unit/backend/helpers/celebrate'
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+
+import { DatabaseError } from '../../core/core.errors'
+import { ExamplesFactory } from '../examples.factory'
+import { ExamplesRouter } from '../examples.routes'
+import { getExampleForms, getSingleExampleForm } from '../examples.service'
+import { RetrievalType } from '../examples.types'
+
+import prepareTestData, {
+  SearchTerm,
+  TestData,
+} from './helpers/prepareTestData'
+
+// Mock min sub count so anything above 0 submissions will be counted.
+jest.mock('../examples.constants', () => ({
+  PAGE_SIZE: 16,
+  MIN_SUB_COUNT: 0,
+}))
+
+jest.mock('../examples.factory')
+const MockExamplesFactory = mocked(ExamplesFactory)
+
+const app = setupApp('/examples', ExamplesRouter, {
+  setupWithAuth: true,
+})
+
+describe('examples.routes', () => {
+  let request: Session
+  let orgTestData: TestData
+  let comTestData: TestData
+  let defaultUser: IUserSchema
+  let comAgency: IAgencySchema
+  let orgAgency: IAgencySchema
+
+  beforeAll(async () => {
+    await dbHandler.connect()
+    const orgData = await dbHandler.insertFormCollectionReqs({
+      mailDomain: 'example.org',
+      shortName: 'orgTest',
+    })
+    orgAgency = orgData.agency
+    defaultUser = orgData.user
+
+    const comData = await dbHandler.insertFormCollectionReqs({
+      mailDomain: 'example.com',
+      shortName: 'comTest',
+    })
+    comAgency = comData.agency
+
+    // Populate database with both users' forms.
+    orgTestData = await prepareTestData(orgData.user, orgData.agency)
+    comTestData = await prepareTestData(comData.user, comData.agency)
+  })
+  beforeEach(async () => {
+    request = supertest(app)
+  })
+  afterEach(async () => {
+    jest.clearAllMocks()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('GET /examples', () => {
+    describe('AggregateStats feature enabled', () => {
+      beforeAll(() => {
+        MockExamplesFactory.getExampleForms.mockImplementation(
+          getExampleForms(RetrievalType.Stats),
+        )
+      })
+
+      it('should return 200 with array of example forms for all agencies when query.agency is missing', async () => {
+        // Arrange
+        const session = await createAuthedSession(defaultUser.email, request)
+
+        // Act
+        const response = await session.get('/examples').query({
+          pageNo: '0',
+        })
+
+        // Assert
+        // Should have both comTestData and orgTestData since searching by all
+        // agencies.
+        const expectedBody = keyBy(
+          [
```

Comment:
minor nit but can we define the elements of this array as constants? otherwise v hard to read

Reply:
no worries, thanks for making me refactor, looks cleaner. Refactored in 2dd6606